components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/APIProviderHostObject.java

```diff
@@ -426,6 +435,99 @@ public static NativeArray jsFunction_getBusinessOwners(Context cx, Scriptable th
+                                                           Function funObj) throws AppManagementException {
+        if (args == null || args.length != 1) {
+            handleException("Invalid number of input parameters.");
+        }
+        if (args[0] == null || args[0] == "null") {
+            handleException("Error while reading business owner. Owner id is null");
+        }
+        NativeArray myn = new NativeArray(0);
+        int ownerId = Integer.valueOf(args[0].toString());
+        APIProvider apiProvider = getAPIProvider(thisObj);
+        BusinessOwner businessOwner = apiProvider.getBusinessOwner(ownerId);
+        NativeObject row = new NativeObject();
+        row.put("businessOwnerId", row, businessOwner.getBusinessOwnerId());
+        row.put("businessOwnerId", row, businessOwner.getBusinessOwnerId());
+        row.put("businessOwnerName", row, businessOwner.getBusinessOwnerName());
+        row.put("businessOwnerEmail", row, businessOwner.getBusinessOwnerEmail());
+        row.put("businessOwnerDescription", row, businessOwner.getBusinessOwnerDescription());
+        row.put("businessOwnerSite", row, businessOwner.getBusinessOwnerSite());
+        List<BusinessOwnerProperties> businessOwnerPropertiesList = businessOwner.getBusinessOwnerPropertiesList();
+        if(businessOwnerPropertiesList != null) {
+            JSONObject businessOwnerPropertiesObject = new JSONObject();
+            for (int i = 0; i < businessOwnerPropertiesList.size(); i++) {
+                JSONObject businessOwnerPropertyObject = new JSONObject();
+                BusinessOwnerProperties businessOwnerProperties = businessOwnerPropertiesList.get(i);
+                businessOwnerPropertyObject.put("propertyValue", businessOwnerProperties.getPropertyValue());
+                businessOwnerPropertyObject.put("isShowingInStore", businessOwnerProperties.isShowingInStore());
+                businessOwnerPropertiesObject.put(businessOwnerProperties.getPropertyId(),businessOwnerPropertyObject);
+            }
+            row.put("businessOwnerProperties", row, businessOwnerPropertiesObject.toJSONString());
+        } else {
+            row.put("businessOwnerProperties", row, null);
+        }
+        return row;
+    }
+
+    /**
+     * Get business owners with pagination.
+     * @param cx
+     * @param thisObj
+     * @param args
+     * @param funObj
+     * @return
+     * @throws AppManagementException
+     */
+    public static NativeObject jsFunction_getBusinessOwnersWithPagination(Context cx, Scriptable thisObj, Object[] args,
+                                                           Function funObj) throws AppManagementException {
+        NativeArray myn = new NativeArray(0);
+        APIProvider apiProvider = getAPIProvider(thisObj);
+        int startIndex = Integer.parseInt(args[0].toString());
+        int pageSize = Integer.parseInt(args[1].toString());
+        int currentPage = Integer.parseInt(args[2].toString());
```

Comment:
Please validate the arguments. Otherwise, there can be ArrayIndexOutOfBoundsException.